src/include/txn.i

```diff
@@ -786,11 +835,22 @@ static inline int
 __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 {
 	WT_TXN *txn;
+	uint32_t ignore_prepare_set;
 
 	txn = &session->txn;
+	ignore_prepare_set = F_ISSET(txn, WT_TXN_IGNORE_PREPARE);
+	/*
+	 * Clear the ignore prepare setting of txn, as it is not suppose to
+	 * affect the visibility for update operation
+	 */
+	if (ignore_prepare_set)
+		F_CLR(txn, WT_TXN_IGNORE_PREPARE);
+
 	if (txn->isolation == WT_ISO_SNAPSHOT)
 		while (upd != NULL && !__wt_txn_upd_visible(session, upd)) {
 			if (upd->txnid != WT_TXN_ABORTED) {
+				if (ignore_prepare_set)
+					F_SET(txn, WT_TXN_IGNORE_PREPARE);
```

Comment:
It might also be cleaner to pass a boolean `ignore_prepare` into `__wt_txn_upd_visible_type()` instead of checking `F_ISSET(txn, WT_TXN_IGNORE_PREPARE)`. If `__wt_txn_update_check()` called `__wt_txn_upd_visible_type()` directly, then it wouldn't have to  set/clear the `txn` flags at all.

Reply:
I haven't made this change, as this is resulting in adding WT_TXN_IGNORE_PREPARE to rec_write.c and bt_ret.c. 
If you feel strongly about this change, I will do it.